### ISSUE TITLE
feat[dev]: ✨ implement A11y Re-Improvements to sd-expandable

### ISSUE THREAD

**Issue by yoezlem** [solid-design-system/solid#2051]
## Description
This tasks describes the Figma implementation for: #2048 

This task describes the Storybook implementation for the sd-expandable component as part of the A11y re-improvements outlined in Epic #2048.

A11y improvements for sd-expandable can be found here: [sd-expandable Notes](https://www.notion.so/matuzo/sd-expandable-1baf3f6e1bc081c3a6c0f66d008bdb50).

------------------------------------------

## Comment
**Problem:** The content isn’t properly hidden. The example works because there are no interactive elements inside the content area. When you add links, they will be accessible to keyboard users and also screen reader users who use the Tab key to navigate.
**Solution:** Instead of toggling `aria-hidden`, use the `inert` attribute.

## Notes
- It's unclear why the content is wrapped in a details element with a hidden summary. This could also be a div element.

## Accessibility information
- Labels should be unique.

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch
